### PR TITLE
Created GetPackageGraphs Function

### PR DESCRIPTION
**Pull request by bburns632**
I created the `GetPackageGraphs` function to allow a user direct access to the `igraph` network objects without creating the html report.  However, this function and `CreatePackageReport` share about 90% of their workflow. 

Since `GetPackageGraphs` and `CreatePackageReport` shared so much functionality, I created a third function `.BuildPackageReporters` to encapsulate that code.   

Unit tests also created.

### CODE REVIEW

File: R/CreatePackageReport.R [hunk 1]
```diff
@@ -63,6 +37,41 @@ CreatePackageReport <- function(packageName
 }
 
 
+#' @title Get The Graph Objects From The Package
+#' @name GetPackageGraphs
+#' @author B. Burns
+#' @description This function will extract and return the graph structues that represent
```
Comment (jameslamb) [inline R/CreatePackageReport.R]: typo. structues --> structures

File: R/CreatePackageReport.R [hunk 2]
```diff
@@ -63,6 +37,41 @@ CreatePackageReport <- function(packageName
 }
 
 
+#' @title Get The Graph Objects From The Package
+#' @name GetPackageGraphs
+#' @author B. Burns
+#' @description This function will extract and return the graph structues that represent 
+#' the functions within the package and the depencies external to the package.  No HTML report
```
Comment (jameslamb) [inline R/CreatePackageReport.R]: typo: depencies --> dependencies

**Comment by bburns632** [comment]
@jameslamb , when you're right, you're right.  

The whole point is to more easily expose the graph object to a user, but with a tagging on some names to the output list object, I think that is accomplished easily enough.

Let's say I'm looking to consolidate the functions in my beefy R package.  One step might be to examine the network structure of the functions to see which functions have the same or very similar dependencies.  Here's an example: 

```library(pkgnet)
reporterList <- CreatePackageReport("lubridate")
g <- reporterList[["PackageFunctionReporter"]]$make_graph_object()
sim <- igraph::similarity(g)
diag(sim) <- 0
which(sim > 0.99, arr.ind = TRUE)
```

**Comment by jameslamb** [comment]
@bburns632 I like your example code!  That's a cool idea. You should add an example like that to the README

**Pull request merged**